### PR TITLE
Do not invoke superclass ctor when no primary ctor.

### LINKED ISSUE #223: Fix superclass emission when primary constructor is missing
Test case:

```kotlin
@Test fun classExtendsNoPrimaryConstructor() {
  val typeSpec = TypeSpec.classBuilder("IoException")
      .superclass(Exception::class)
      .addFunction(FunSpec.constructorBuilder().build())
      .build()

  assertThat(toString(typeSpec)).isEqualTo("""
      |package com.squareup.tacos
      |
      |import java.lang.Exception
      |
      |class IoException : Exception() {
      |  constructor() {
      |  }
      |}
      |""".trimMargin())
}
```

The output fails to compile on the `Exception()` part with `Supertype initialization is impossible without primary constructor`. Correct syntax is `Exception` without braces.

### PR DESCRIPTION
**Pull request by JakeWharton**
And when secondary ctors are present.

Closes #223.

### CODE REVIEW

File: src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt [hunk 1]
```diff
@@ -2498,6 +2498,51 @@ class TypeSpecTest {
...
+      |""".trimMargin())
+  }
+
+  @Test fun classExtendsNoPrimaryConstructorButSuperclassParams() {
+    try {
+      TypeSpec.classBuilder("IoException")
+          .superclass(Exception::class)
+          .addSuperclassConstructorParameter("%S", "hey")
+          .addFunction(FunSpec.constructorBuilder().build())
+          .build()
```
Comment (vanniktech) [inline src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: Missing a fail

Comment (JakeWharton) [reply src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: Bah we need a lambda version of this assertion so I don't have to bother

File: src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt [hunk 2]
```diff
@@ -2498,6 +2498,52 @@ class TypeSpecTest {
...
+      fail()
+    } catch (e: IllegalArgumentException) {
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: nit: `expected`

Comment (JakeWharton) [reply src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: We usually only use that name when the value is unused in the `catch` block. Another reason to move to a `assertThrows` helper.

**Pull request merged**